src/storage/mod.rs

```diff
@@ -0,0 +1,337 @@
+        let mut storage = Storage::open(storage_path.as_str(), true).unwrap();
+
+        assert!(storage.set_column_family("cf1").is_ok());
+        assert!(storage.set_column_family("cf2").is_ok());
+
+        let (cf1, cf2) = (
+            storage.get_column_family("cf1").unwrap(),
+            storage.get_column_family("cf2").unwrap()
+        );
+
+        let tx = storage.create_transaction().unwrap();
+
+        assert!(tx.is_empty_cf(cf1).unwrap() && tx.is_empty_cf(cf2).unwrap());
+
+        // 1-st update -------------------------------------------------------------------
+        tx.update_cf(cf1,
+                     &vec![
+                            ("k11".as_ref(), "v11".as_ref())],
+                     &vec![]).unwrap();
+        tx.save().unwrap();
+        // 2-nd update -------------------------------------------------------------------
+        tx.update_cf(cf2,
+                     &vec![
+                            ("k21".as_ref(), "v21".as_ref())],
+                     &vec![]).unwrap();
+        tx.save().unwrap();
+        // 3-rd update -------------------------------------------------------------------
+        tx.update_cf(cf1,
+                     &vec![
+                            ("k12".as_ref(), "v12".as_ref())],
+                     &vec![]).unwrap();
+        tx.update_cf(cf2,
+                     &vec![
+                            ("k22".as_ref(), "v22".as_ref())],
+                     &vec![]).unwrap();
+        tx.save().unwrap();
+        // 4-th update -------------------------------------------------------------------
+        tx.update_cf(cf2,
+                     &vec![
+                            ("k23".as_ref(), "v23".as_ref())],
+                     &vec![]).unwrap();
+
+        // State after all updates
+        assert_eq!(tx.get_cf(cf1, b"k11").unwrap(), b"v11");
+        assert_eq!(tx.get_cf(cf1, b"k12").unwrap(), b"v12");
+        assert_eq!(tx.get_cf(cf2, b"k21").unwrap(), b"v21");
+        assert_eq!(tx.get_cf(cf2, b"k22").unwrap(), b"v22");
+        assert_eq!(tx.get_cf(cf2, b"k23").unwrap(), b"v23");
+
+        // Discarding the 4-th update
+        tx.rollback_to_savepoint().unwrap();
+        assert_eq!(tx.get_cf(cf1, b"k11").unwrap(), b"v11");
+        assert_eq!(tx.get_cf(cf1, b"k12").unwrap(), b"v12");
+        assert_eq!(tx.get_cf(cf2, b"k21").unwrap(), b"v21");
+        assert_eq!(tx.get_cf(cf2, b"k22").unwrap(), b"v22");
+        assert!(tx.get_cf(cf2, b"k23").is_none());
+
+        // Discarding the 3-rd update
+        tx.rollback_to_savepoint().unwrap();
+        assert_eq!(tx.get_cf(cf1, b"k11").unwrap(), b"v11");
+        assert_eq!(tx.get_cf(cf2, b"k21").unwrap(), b"v21");
+        assert!(tx.get_cf(cf1, b"k12").is_none());
+        assert!(tx.get_cf(cf2, b"k22").is_none());
+        assert!(tx.get_cf(cf2, b"k23").is_none());
+
+        // Discarding all the updates since the transaction was created (the 1-st and the 2-nd updates)
+        tx.rollback().unwrap();
```

Comment:
Yes, when a transaction is committed there are no savepoints, so it can't be rolled back to any savepoint with rollback_to_savepoint() as well as it can't be completely rolled back with rollback(). I've added this scenario to the tests